### DISCUSSION TITLE
Is there a way to use the typechain typings to get encoded calldata?

### DISCUSSION THREAD

**Discussion by stevieraykatz** [ethereum-ts/TypeChain#820] [Q&A] [answered]
Hi hi, I think I'm _so close_ to having this but can't seem to get anything to work. 

I'm trying to generate a `data` field for a multisig transaction proposal which would be the calldata associated with the desired MS transaction. The method I'm attempting to encode into the data field is actually one of the multisig's own methods. So! I have something like this: 

```
const multisig = await hre.ethers.getContractAt("MultiSigGeneric",multisigAddress) as MultiSigGeneric
let addOwnerCalldata = await multisig.interface.functions.encode.addOwner(newOwner.address) // this syntax is broken
await multisig.submitTransaction( 
    "add owner",  //title
    "add newOwner as owner", //description
    multisig.address,  //destination,
    0,  //value
    addOwnerCalldata  //data
)
```

can someone help me with the syntax for line 2 using the typings? All of the ways I can find around use the ABI wrapping method which seems redundant when we already have the function abi parsed into types.

**Comment by zemse** [comment, marked as answer]
You can use the populateTransaction
```ts
// ethers v5
const {data} = await multisigContract.populateTransaction.addOwner(newOwner)
// ethers v6
const {data} = await multisigContract.addOwner.populateTransaction(newOwner)
```

**Reply by stevieraykatz** [reply]
Excellent!! Thank you
